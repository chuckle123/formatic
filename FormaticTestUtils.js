import React from 'react';
import { mount } from 'enzyme';

import Formatic from './lib/formatic';

export const renderFieldsToHtml = fields => {
  const config = Formatic.createConfig();
  const value = config.createRootValue({
    fields,
  });

  const component = mount(
    <Formatic
      config={config}
      value={value}
      fields={fields}
      onChange={() => {}}
    />
  );
  const html = component.html();
  component.unmount();
  return html;
};
