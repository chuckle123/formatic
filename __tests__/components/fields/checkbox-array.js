/*global describe, it, expect*/

import { renderFieldsToHtml } from '../../../FormaticTestUtils';
import checkboxArrayExample from '../../../demo/examples/checkbox-array';

describe('checkbox-array field', () => {
  it('should render correctly', () => {
    expect(renderFieldsToHtml(checkboxArrayExample.fields)).toMatchSnapshot();
  });
});
