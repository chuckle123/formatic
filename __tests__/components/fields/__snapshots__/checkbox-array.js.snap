// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox-array field should render correctly 1`] = `

<div class="formatic">
  <div class="optional">
    <div class>
      <span class="required-label not-required-text">
      </span>
    </div>
    <span>
    </span>
    <fieldset class>
      <div class="optional">
        <div class>
          <label>
            Colors
          </label>
          <span class="required-label not-required-text">
          </span>
        </div>
        <span>
        </span>
        <div class>
          <span class="field-choice">
            <span style="white-space: nowrap;">
              <input name="colors"
                     type="checkbox"
                     value="red"
              >
              <span>
              </span>
              <span class="field-choice-label">
                Red
              </span>
            </span>
          </span>
          <span class="field-choice">
            <span style="white-space: nowrap;">
              <input name="colors"
                     type="checkbox"
                     value="green"
              >
              <span>
              </span>
              <span class="field-choice-label">
                Green
              </span>
            </span>
          </span>
          <span class="field-choice">
            <span style="white-space: nowrap;">
              <input name="colors"
                     type="checkbox"
                     value="blue"
              >
              <span>
              </span>
              <span class="field-choice-label">
                Blue
              </span>
            </span>
          </span>
        </div>
      </div>
      <div class="optional readonly">
        <div class>
          <label>
            Readonly Colors
          </label>
          <span class="required-label not-required-text">
          </span>
        </div>
        <span>
        </span>
        <div class>
          <span class="field-choice">
            <span style="white-space: nowrap;">
              <input name="readonly-colors"
                     type="checkbox"
                     disabled
                     value="red"
              >
              <span>
              </span>
              <span class="field-choice-label">
                Red
              </span>
            </span>
          </span>
          <span class="field-choice">
            <span style="white-space: nowrap;">
              <input name="readonly-colors"
                     type="checkbox"
                     disabled
                     value="green"
              >
              <span>
              </span>
              <span class="field-choice-label">
                Green
              </span>
            </span>
          </span>
          <span class="field-choice">
            <span style="white-space: nowrap;">
              <input name="readonly-colors"
                     type="checkbox"
                     disabled
                     value="blue"
              >
              <span>
              </span>
              <span class="field-choice-label">
                Blue
              </span>
            </span>
          </span>
        </div>
      </div>
      <div class="optional">
        <div class>
          <label>
            Colors
          </label>
          <span class="required-label not-required-text">
          </span>
        </div>
        <span>
        </span>
        <div class>
          <span class="field-choice">
            <span style="white-space: nowrap;">
              <input name="colors-alt"
                     type="checkbox"
                     value="blue"
              >
              <span>
              </span>
              <span class="field-choice-label">
                Blue
              </span>
            </span>
          </span>
          <span class="field-choice">
            <span style="white-space: nowrap;">
              <input name="colors-alt"
                     type="checkbox"
                     value="yellow"
              >
              <span>
              </span>
              <span class="field-choice-label">
                Yellow
              </span>
            </span>
          </span>
          <span class="field-choice">
            <span style="white-space: nowrap;">
              <input name="colors-alt"
                     type="checkbox"
                     value="purple"
              >
              <span>
              </span>
              <span class="field-choice-label">
                Purple
              </span>
            </span>
          </span>
        </div>
      </div>
    </fieldset>
  </div>
</div>

`;
